daemon/api_users_test.go

```diff
@@ -837,6 +838,47 @@ func (s *userSuite) TestPostCreateUserFromAssertionAllKnownButOwned(c *check.C)
 	c.Check(rsp.Result, check.DeepEquals, expected)
 }
 
+func (s *userSuite) TestPostCreateUserAutomaticDisabled(c *check.C) {
+	s.makeSystemUsers(c, []map[string]interface{}{goodUser})
+
+	// disable automatic user creation
+	st := s.d.Overlord().State()
+	st.Lock()
+	tr := config.NewTransaction(st)
+	err := tr.Set("core", "users.create.automatic", false)
+	tr.Commit()
+	st.Unlock()
+	c.Assert(err, check.IsNil)
+
+	defer daemon.MockOsutilAddUser(func(username string, opts *osutil.AddUserOptions) error {
+		panic("no user should be created")
+	})()
+	// make sure we report them as non-existing until created
+	defer daemon.MockUserLookup(func(username string) (*user.User, error) {
+		return nil, fmt.Errorf("not created yet")
```

Comment:
(nitpick) Unless I misread/misremember this should also not get called so we could panic() here too maybe?
E.g.
```
--- a/daemon/api_users_test.go
+++ b/daemon/api_users_test.go
@@ -855,7 +855,8 @@ func (s *userSuite) TestPostCreateUserAutomaticDisabled(c *check.C) {
        })()
        // make sure we report them as non-existing until created
        defer daemon.MockUserLookup(func(username string) (*user.User, error) {
-               return nil, fmt.Errorf("not created yet")
+               panic("no lookup is needed because no user will be created")
+               return nil, fmt.Errorf("should not be called")
        })()
 
        // do it!
```

Reply:
it's not the essential bit of creating users though, so I think it's fine as is

Reply:
Sure, fine as well